rcl_lifecycle/src/rcl_lifecycle.c

```diff
@@ -163,14 +165,14 @@ rcl_lifecycle_trigger_transition(
 
   // If we have a faulty transition pointer
   if (!transition) {
-    fprintf(stderr, "No transition found for node %s with key %d\n",
-      state_machine->current_state->label, key);
+    RCUTILS_LOG_ERROR("No transition found for node %s with key %d",
+      state_machine->current_state->label, key)
```

Comment:
Nitpicks: some macros have `;` after the closing parenthesis and some don't. I would expect (according to how the macros are implemented) to either fail to compile without the semi-colon or to produce a pedantic warning if they present and not needed